pkg/operation/botanist/health_check.go

```diff
@@ -816,6 +813,31 @@ func (b *Botanist) healthChecks(initializeShootClients func() error, thresholdMa
 	return apiserverAvailability, controlPlane, nodes, systemComponents
 }
 
+/*This function is created because when Loki is deployed for the first time on already existing shoot
+the maintenance window will not allow Loki StatfulSet to be deployed and the health check will fail.
+Do we are preventing this health check up to a certain version when Loki will be still integrated to the
+landscapes. In versions above 1.8.0 this function can be removed */
+func (b *Botanist) isLoggingHealthCheckNeeded() bool {
+	if b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting {
+		return false
+	}
+	gardenerVersion18, err := semver.NewVersion("1.8.0")
+	if err != nil {
+		return false
+	}
+
+	lastGardenerVersion, err := semver.NewVersion(b.Shoot.Info.Status.Gardener.Version)
```

Comment:
`b.Shoot.Info.Status.Gardener` can be nil
`b.Shoot.Info.Status.Gardener.Version` can be nil.

Reply:
could you double check it. I see Gardener as a struct and Version as a string

Reply:
shoot and  Info could be nill

Reply:
You are right, sorry for the misleading comment.